### PR TITLE
cargo-ndk: init at 2.12.6

### PR DESCRIPTION
**Pull request by MGlolenstine**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/rust/cargo-ndk/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    owner = "bbqsrc";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-VGdFIMyZhb7SDWAXgs7kFlblYCO4rLf+3/N7Mashc4o=";
+  };
+
+  cargoHash = "sha256-pv6t9oKj5tdQjpvBgj/Y11uKJIaIWcaA9ib/Id/s+qs=";
+
+  buildInputs = lib.optionals stdenv.isLinux [ glibc ]
+    ++ lib.optionals stdenv.isDarwin [ CoreServices Security ];
```
Comment (NickCao) [inline pkgs/development/tools/rust/cargo-ndk/default.nix:27]:
```suggestion
    ++ lib.optionals stdenv.isDarwin (with darwin.apple_sdk.frameworks;[ CoreServices Security ]);
```

File: pkgs/development/tools/rust/cargo-ndk/default.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+, rustPlatform
+, fetchFromGitHub
+, pkg-config
+, glibc
+, darwin
+}:
+
+let
+  inherit (darwin.apple_sdk.frameworks) CoreServices Security;
+in
```
Comment (NickCao) [inline pkgs/development/tools/rust/cargo-ndk/default.nix:12]:
```suggestion
```

File: pkgs/development/tools/rust/cargo-ndk/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-VGdFIMyZhb7SDWAXgs7kFlblYCO4rLf+3/N7Mashc4o=";
+  };
+
+  cargoHash = "sha256-pv6t9oKj5tdQjpvBgj/Y11uKJIaIWcaA9ib/Id/s+qs=";
+
+  buildInputs = lib.optionals stdenv.isLinux [ glibc ]
+    ++ lib.optionals stdenv.isDarwin (with darwin.apple_sdk.frameworks;[ CoreServices Security ]);
+  nativeBuildInputs = [ pkg-config ];
```
Comment (NickCao) [inline pkgs/development/tools/rust/cargo-ndk/default.nix:25]: Isn't glibc already part of stdenv? Also please order nativeBuildInputs before buildInputs for aesthetic reasons.

Comment (MGlolenstine) [reply pkgs/development/tools/rust/cargo-ndk/default.nix]: I'm not sure if it is, but removing it doesn't break the build, so I don't think it's needed :)

File: pkgs/development/tools/rust/cargo-ndk/default.nix [hunk 4]
```diff
@@ -0,0 +1,34 @@
+{ lib
+, stdenv
+, rustPlatform
+, fetchFromGitHub
+, pkg-config
+, glibc
```
Comment (NickCao) [inline pkgs/development/tools/rust/cargo-ndk/default.nix:6]: Remove the no longer needed argument.

File: pkgs/development/tools/rust/cargo-ndk/default.nix [hunk 5]
```diff
@@ -0,0 +1,34 @@
+{ lib
+, stdenv
+, rustPlatform
+, fetchFromGitHub
+, pkg-config
```
Comment (NickCao) [inline pkgs/development/tools/rust/cargo-ndk/default.nix:5]: Maybe also remove pkg-config? Please try.

Comment (MGlolenstine) [reply pkgs/development/tools/rust/cargo-ndk/default.nix:5]: It does work, as there's no native dependencies anymore :)

**Comment by NickCao** [comment]
And, build is failing on darwin.
```
note: ld: framework not found CoreGraphics
```

**Comment by MGlolenstine** [comment]
I'm not entirely sure how darwin works.
I'll try removing the optional requirements and will check to see if it builds without them.

**Comment by NickCao** [comment]
Another error jumps out...
`note: ld: framework not found Foundation`
If you don't have access to a darwin machine, another choice would be simply setting
`platforms = platform.linux`
And throw darwin support away. We can always add that later.

**Comment by MGlolenstine** [comment]
That is a fair point, let's do that then and we'll burn that bridge when we come to it.

**Review by NickCao — APPROVED** [review]

**Pull request merged**